pkg/streamingpromql/engine_test.go

```diff
@@ -1810,3 +1767,78 @@ func TestCompareVariousMixedMetrics(t *testing.T) {
 		}
 	}
 }
+
+func TestCompareVariousMixedMetricsBinaryOperations(t *testing.T) {
+	labelsToUse := []string{"a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o"}
```

Comment:
As far as I can tell, this is the same for each test case - could we move this to one place?

For bonus points: could we somehow generate this from the series definitions in `runMixedMetricsTests` or vice versa, so they never get out of sync?

Reply:
This is needed because each set of tests may want to do different labels. I can deduplicate it a bit though, but I'm not aiming for bonus points ;-).